addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/internal/servelet/KonnectedHTTPServelet.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.konnected.internal.servelet;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.openhab.binding.konnected.handler.KonnectedHandler;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Main OSGi service and HTTP servlet for Konnected Webhook.
+ *
+ * @author Zachary Christiansen - Initial contribution
+ */
+public class KonnectedHTTPServelet extends HttpServlet {
+    private static final long serialVersionUID = 1288539782077957954L;
+    private static final String PATH = "/Konnected";
+    private static final String APPLICATION_JSON = "application/json";
+    private static final String CHARSET = "utf-8";
+    private final Gson gson = new Gson();
+
+    private final Logger logger = LoggerFactory.getLogger(KonnectedHTTPServelet.class);
+
+    private HttpService httpService;
+    private KonnectedHandler bridgeHandler;
+    private String path;
+
+    public KonnectedHTTPServelet(HttpService httpService, String id) {
+        this.httpService = httpService;
+        this.path = PATH + "/" + id;
+    }
+
+    /**
+     * OSGi activation callback.
+     *
+     * @param config Service config.
+     **/
+    public void activate(KonnectedHandler bridgeHandler) {
```

Comment:
Please rename the parameter `bridgeHandler` to `thingHandler` because the term "bridge" is already reserved for special things in openHAB. Actually I got confused why you have a bridgehandler but do not define one in your thing-types file .)

Reply:
this is a case of overly aggressive "copying" from other working bindings i certainly understand the confusion and will rename the variable accordingly